pkg/agent/secondarynetwork/podwatch/controller_test.go

```diff
@@ -1147,11 +1147,19 @@ func TestPodControllerAddPod(t *testing.T) {
 		tmpPodNamespace := "testns"
 		tmpPodKey := podKeyGet(tmpPodName, tmpPodNamespace)
 		_, err := podController.assignSriovVFDeviceID(tmpPodName, tmpPodNamespace, sriovResourceName1, interfaceName)
+		require.NoError(t, err, "error while assigning unused VfDevice ID")
 		vfDeviceIDInfoCache, exists := podController.vfDeviceIDUsageMap.Load(tmpPodKey)
 		assert.True(t, exists)
 		vfDeviceIDsInfo := vfDeviceIDInfoCache.([]podSriovVFDeviceIDInfo)
-		assert.Equal(t, interfaceName, vfDeviceIDsInfo[0].ifName, "incorrect interface name")
-		require.NoError(t, err, "error while assigning unused VfDevice ID")
+		var targetEntry *podSriovVFDeviceIDInfo
+		for i := range vfDeviceIDsInfo {
+			if vfDeviceIDsInfo[i].resourceName == sriovResourceName1 {
+				targetEntry = &vfDeviceIDsInfo[i]
+				break
+			}
+		}
+		assert.NotNil(t, targetEntry, "target resource entry should not be nil")
```

Comment:
should be `require.NotNil`, otherwise if this assertion fails the next one will panic

or you can probably just use `assert.Contains` instead of the `for` loop and these 2 assertions?

Reply:
Good idea. Done.